### PR TITLE
[Artist] Add support for formatted works count string

### PR DESCRIPTION
**Pull request by mennenia**
Added a property to artist for a formatted works string, as required for [this issue](https://github.com/artsy/eigen/issues/1760)

- An artist with 50 published works and 28 for sale, it would return "50 works, 28 for sale". 
- An artist with 18 works, none for sale: "18 works".
- An artist with one work for sale: "1 work, 1 for sale". 

I'm having trouble writing tests however. Currently there's an artist set up, up-front, used by each test. 

- How do you stub a dummy artists with properties like their published works? 
- How do you do so just for your own test (not all of the tests in the file)? 
- How can you do it in a way that you can change them between tests?

I noticed the [promises of the artist count](https://github.com/artsy/metaphysics/blob/master/test/schema/artist/index.js#L19), however ideally I'd write various tests to check formatting for artists with no works, only one, etc etc.

This is similar to an issue I worked on with @sarahscott (for formatted nationality and birthdate strings), and @alloy also helped debug that, but we didn't manage.

Any help / tips would be much appreciated!

### CODE REVIEW

File: schema/artist/index.js [hunk 1]
```diff
@@ -238,6 +238,14 @@ const ArtistType = new GraphQLObjectType({
           gravity(`artist/${id}/artworks`, options)
             .then(exclude(options.exclude, 'id')),
       },
+      formatted_artworks_count_string: {
```
Comment (alloy) [inline schema/artist/index.js]: Is the `_string` suffix used elsewhere? I’m generally not a fan of Hungarian notation, in this case especially as the schema specifies it is a string.

**Comment by alloy** [comment]
@mennenia Added an initial test that should show you how to add tests for the rest.

File: schema/artist/index.js (hunk 1 above)
Comment (mennenia) [inline schema/artist/index.js]: fixed!

**Comment by mennenia** [comment]
Tests done, ready for review! :)

**Comment by alloy** [comment]
Great job, very comprehensive 👍 :100:

File: schema/artist/index.js [hunk 2]
```diff
@@ -238,6 +238,18 @@ const ArtistType = new GraphQLObjectType({
           gravity(`artist/${id}/artworks`, options)
             .then(exclude(options.exclude, 'id')),
       },
+      formatted_artworks_count: {
```
Comment (mzikherman) [inline schema/artist/index.js]: super minor but you can do string interpolation (like ruby's `#{..}`):

``` js
totalWorks = `${published_artworks_count} ${published_artworks_count > 1 ? 'works' : 'work')}`
const forSaleWorks = forsale_artworks_count ? `${forsale_artworks_count} for sale` : null
return (forSaleWorks && totalWorks) ? `${totalWorks}, ${forSaleWorks}` : totalWorks;
```

that's actually not really any better now that i look at it.

Comment (mzikherman) [inline schema/artist/index.js]: nvm!

**Comment by alloy** [comment]
@mennenia Can you add a `description` entry to this field that explains what it is and when to use it?

**Comment by sarahscott** [comment]
Pending `description` entity 😄

File: schema/artist/index.js (hunk 2 above)
Comment (mennenia) [inline schema/artist/index.js]: no prob, thanks for having a look!

**Comment by sarahscott** [comment]
👌

**Pull request merged**